text/0014-configuration.md

```diff
@@ -0,0 +1,78 @@
+# Configuration
+- Start Date: 2019/09/09
+- RFC PR: 
+
+## Definitions
+
+**Config files** allow implementers to customize the behavior of microfrontends
+by providing static files. Common formats include JSON, YAML, and HCL.
+
+A **config schema** is the set of expectations that a microfrontend has of
+implementation-provided config files. It says how the config file must be 
+structured, what keys it must have, and constrains what types of values are allowed.
+
+A **config library** is a code module that allows developers to define config 
+schemas, and loads config files in accordance with that schema. It may also 
+provide other tooling related to configuration.
+
+
+## Decision
+
+Microfrontend developers should use the official OpenMRS MF Config Library. 
+The Config Library should be a JavaScript module that
+*   Provides an API for modules to define a config schema
+*   Provides an API for modules access config values
+*   Looks for config files in an implementation’s esm-root-config (or using the Import Map)
```

Comment:
This is a big open question. Do we expect config files to be specified by esm-root-config or by the import map?

Reply:
I prefer root config to be part of esm-root-config mainly because of 1 reason: version control. The import map is currently not versioned so it will be difficult to track the changes overtime to the config files url. The root config is versioned and would be the better alternative.

Reply:
The config file wouldn't be contained in the import map, it would be referred to by the import map. It could be managed in version control just like anything else the import map refers to.

Reply:
I'm open to either putting the implementation config stuff discussed in this RFC into `@openmrs/esm-root-config` or into a new module called `@openmrs/esm-feature-config` (or similar).

JJ, Brandon, and I talked about it for a while on slack a couple days -- I think I lean towards a new module called `@openmrs/esm-feature-config` so that esm-root-config doesn't continue growing into a catch-all for all important features. The "root-config" refers to the "single-spa root config", as described [in the single-spa docs](https://single-spa.js.org/docs/configuration.html).

One concern brought up regarding the creation of a module for this is that it would require distros to have/maintain one more code repo. I don't view this as a big problem - setting up a distribution/implementation will require maintaining a package.json and a couple repos. And installing dependencies from npm and creating an import map, which essentially will just be an `npm install && npm run build`.

Open to other ideas on this, though.